stdlib/public/core/Result.swift

```diff
@@ -89,6 +89,21 @@ public enum Result<Success, Failure: Error> {
   /// Returns a new result, mapping any success value using the given
   /// transformation and unwrapping the produced result.
   ///
+  /// Use this method when you need to transform the value of a `Result`
+  /// instance eventhough it produces a nested result type.
+  ///
+  /// In this example, note the difference in the result of using `map` and
+  /// `flatMap` with a transformation that returns an result type.
+  ///
+  ///     func getNextInteger() -> Result<Int, Error> { /* ... */ }
+  ///     func getNextAfterInteger() -> Result<Int, Error> { /* ... */ }
```

Comment:
Given that these are very simple functions, I think it would be more clear to just define them here completely. Also, `getNextAfterInteger` needs an `Int` parameter:

```suggestion
  ///     func getNextInteger() -> Result<Int, Error> { 
  ///         .success(4) 
  ///     }
  ///     func getNextAfterInteger(_ n: Int) -> Result<Int, Error> {
  ///         .success(n + 1) 
  ///     }
```

Reply:
OK, will send v2 with this change.